faunadb/client.go

```diff
@@ -27,16 +29,27 @@ func Endpoint(url string) ClientConfig { return func(cli *FaunaClient) { cli.end
 // HTTP configures the FaunaClient structure to use a specific http.Client.
 func HTTP(http *http.Client) ClientConfig { return func(cli *FaunaClient) { cli.http = http } }
 
+/*
+EnableTxnTimePassthrough configures the FaunaClient to keep track of the last seen transaction time.
+The last seen transaction time is used to avoid reading stale data from outdated replicas when
+reading and writing from different nodes at the same time.
+*/
+func EnableTxnTimePassthrough() ClientConfig {
+	return func(cli *FaunaClient) { cli.isTxnTimeEnabled = true }
+}
+
 /*
 FaunaClient provides methods for performing queries on a FaunaDB cluster.
 
 This structure should be reused as much as possible. Avoid copying this structure.
 If you need to create a client with a different secret, use the NewSessionClient method.
 */
 type FaunaClient struct {
-	basicAuth string
-	endpoint  string
-	http      *http.Client
+	basicAuth        string
+	endpoint         string
+	http             *http.Client
+	isTxnTimeEnabled bool
+	lastTxnTime      int64
```

Comment:
I wonder if this should be a global variable instead of a per client variable. A global variable would imply contingency but, at the same time, it would be more consistent if you have a connection pool somewhere in your application.

Reply:
I think lets leave it at the client level for now, rather than a global. We can iterate, of course. If we're going to have global state like this we may want to look at having our own connection pools.